src/utils/addressList/index.ts

```diff
@@ -0,0 +1,42 @@
+import { utils as ethersUtils } from "ethers";
+import { AddressListAdapter, INVALID_ADDRESS } from "./types";
+import * as bybit from "./adapters/bybit";
+
+export class AddressList {
+  constructor(readonly addressLists: AddressListAdapter[]) {}
+
+  async update(): Promise<{ length: number; addresses: Set<string> }> {
+    const _addresses = await Promise.all(this.addressLists.map((adapter) => adapter.update()));
+    const addresses = _addresses
+      .flat()
+      .map((address) => {
+        try {
+          return ethersUtils.getAddress(address.toLowerCase());
+        } catch {
+          return INVALID_ADDRESS;
+        }
+      })
+      .filter((address) => address !== INVALID_ADDRESS);
+    return { addresses: new Set(addresses), length: addresses.length };
```

Comment:
Is `length` the size of the array before removing duplicates via `Set` on purpose? Should probably be the size of the set, correct?

Reply:
Good catch on the mismatch there; that's definitely not intended. Length should be the number of elements in the set. The set API is just frustrating in that it seems you have to revert back to an array in order to figure out the length; that's an annoyance I'd hoped to encapsulate here.

Reply:
Fixed here: https://github.com/across-protocol/sdk/pull/902#discussion_r1975751401